### PR TITLE
Arc-2676 connected state

### PR DESCRIPTION
**Pull request by rachellerathbone**
**What's in this PR?**
Connected state component that will be used in both the admin and global pages.

If a pipeline ran within the last 24 hours you'll see the time formatted like:

Screenshot 2023-11-22 at 10 42 15 am

Otherwise, it will look like:

Screenshot 2023-11-22 at 10 42 07 am

**Why**
Coz it wasn't done yet?

**Added feature flags**
All behind RENOVATED_JENKINS_FOR_JIRA_CONFIG_FLOW

**Affected issues**  
Arc-2676

**How has this been tested?**  
Locally and staging

**What's Next?**
* Consume data from the backend, update existing dummy data, and write tests for set up guide (didn't do it when I created the set up guide component as I would have to change the tests once real data was flowing in)
* A loader will be added when I build the connection wizard for the empty state

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import moment from 'moment';
+import '@testing-library/jest-dom/extend-expect';
+import { ConnectedJenkinsServers, timeFromNow } from './ConnectedJenkinsServers';
+import { EventType, JenkinsServer } from '../../../../src/common/types';
+
+describe('timeFromNow util', () => {
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.test.tsx:8]: Thought about setting this up in its own test file but figured it made more sense to go here at this point as it is only being used by this component. If we end up using it elsewhere we can move both the util function and its tests to their own files.

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanel.tsx [hunk 2]
```diff
@@ -4,21 +4,44 @@ import { ConnectionPanelMain } from './ConnectionPanelMain';
 import { ConnectionPanelTop } from './ConnectionPanelTop';
 import { ConnectedState } from '../StatusLabel/StatusLabel';
 import { connectionPanelContainer } from './ConnectionPanel.styles';
+import { JenkinsServer } from '../../../../src/common/types';
+import { getAllJenkinsServers } from '../../api/getAllJenkinsServers';
+
+// TODO - add DUPLICATE state once I'm pulling in new data from backend
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanel.tsx:10]: Added this and other todos to a todo list that I'm tackling as I work through this.

File: app/src/common/types.ts [hunk 3]
```diff
@@ -7,12 +7,19 @@ export enum EventType {
...
+	DUPLICATE = 'DUPLICATE',
+	PENDING = 'PENDING'
+}
+
 export interface JenkinsServer {
 	uuid: string,
 	name: string,
 	secret?: string,
 	pipelines: JenkinsPipeline[],
 	pluginConfig?: JenkinsPluginConfig;
```
Comment (joshkay10) [inline app/src/common/types.ts:21]: existing issue on this line, `pluginConfig?: JenkinsPluginConfig;` can we replace the `; `with a` , `

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx [hunk 4]
```diff
@@ -1,44 +1,73 @@
...
-}) => (
-	<div className={cx(testid === 'setUpGuidePanel'
-		? setUpGuideContainer : connectionPanelMainTabs)} data-testid={testid }>
-		{children}
-	</div>
-);
+}: PanelProps) => {
+	let className;
+
+	if (testid === 'setUpGuidePanel') {
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx:29]: using the testid outside of tests feels strange, but if thats what needed 🤷

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx [hunk 5]
```diff
@@ -1,44 +1,73 @@
 import React, { ReactNode } from 'react';
 import { cx } from '@emotion/css';
 import Tabs, { Tab, TabList, TabPanel } from '@atlaskit/tabs';
-import { connectionPanelMainContainer, connectionPanelMainTabs, setUpGuideContainer } from './ConnectionPanel.styles';
+import {
+	connectionPanelMainContainer,
+	connectionPanelMainConnectedTabs,
+	connectionPanelMainNotConnectedTabs, setUpGuideContainer
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx:7]: :eyes:

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+				},
+				{
+					key: 'time',
+					content: 'Received'
+				}
+			]
+		};
+	};
+
+	const rows = (serverName: string, serverId: string, pipelines: JenkinsPipeline[] = []) => {
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx:54]: /nit function type

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	connectedStateCellContainer, connectedStateCellEvent,
+	connectedStateContainer
+} from './ConnectionPanel.styles';
+
+export const timeFromNow = (date: moment.MomentInput): string => {
+	const seconds = moment().diff(moment(date), 'seconds');
+	const minutes = moment().diff(moment(date), 'minutes');
+	const hours = moment().diff(moment(date), 'hours');
+
+	if (hours > 0) {
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx:21]: big fan of early returns so its nice to see :hyperblob:

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+import { cx } from '@emotion/css';
+import DynamicTable from '@atlaskit/dynamic-table';
+import { JenkinsPipeline, JenkinsServer } from '../../../../src/common/types';
+import {
+	mapLastEventStatus,
+	mapLastEventStatusIcons
+} from '../JenkinsServerList/ConnectedServer/ConnectedServers';
+import {
+	connectedStateCell,
+	connectedStateCellContainer, connectedStateCellEvent,
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx:12]: 👀

File: app/src/common/types.ts (hunk 3 above)
Comment (rachellerathbone) [reply app/src/common/types.ts:21]: Oh yeah, that was stoopid of me wasn't it?

Comment (rachellerathbone) [reply app/src/common/types.ts:21]: Surprised the linter didn't complain coz it shits the bed usually over the smallest things

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx (hunk 8 above)
Comment (rachellerathbone) [reply app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectedJenkinsServers.tsx:12]: God damn IntelliJ and it's sucky atuo import

File: app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx (hunk 4 above)
Comment (rachellerathbone) [reply app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx:29]: lol you know this was in the PR you approved yesterday? 😆 It's just up as green coz I had to refactor to make sure this supported both the ConnectedState tab and the SetUpGuide one. I agree but think our only other alternative is to pass a isSetUpGuide prop from Panel component of SetUpGuide. Thoughts?

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**